src/c/discovery.c

```diff
@@ -164,3 +166,65 @@ void edgex_device_handler_discoveryv2 (void *ctx, const devsdk_http_request *req
     reply->content_type = CONTENT_PLAINTEXT;
   }
 }
+
+static edgex_baseresponse *edgex_disc_delete_response_create (uint64_t code, char *msg, const char * req_id)
+{
+  edgex_baseresponse *res = malloc (sizeof (edgex_baseresponse));
+  res->statusCode = code;
+  if (msg) res->message = msg;
```

Comment:
if it's ok for res->message to be NULL, make this assignment unconditional. Otherwise do `res->message = msg ? msg : ""`